src/state/slices/limitOrderSlice/limitOrderSlice.ts

```diff
@@ -0,0 +1,211 @@
+    },
+    confirmSubmit: (state, action: PayloadAction<QuoteId>) => {
+      const limitOrderQuoteId = action.payload
+
+      // If there is no active quote, the state is corrupt.
+      if (state.activeQuote === undefined) {
+        console.error('Attempted to confirm an non-existent limit order quote')
+        return
+      }
+
+      // If this is not true, the state is corrupt.
+      assert(state.activeQuote.response.id === limitOrderQuoteId)
+
+      const unsignedOrderCreation = buildUnsignedLimitOrder(state.activeQuote)
+      state.confirmedLimitOrder[limitOrderQuoteId] = {
+        params: state.activeQuote.params,
+        unsignedOrderCreation,
+      }
+
+      if (state.orderSubmission[limitOrderQuoteId].state !== LimitOrderSubmissionState.Previewing) {
+        if (
+          state.orderSubmission[limitOrderQuoteId].state === LimitOrderSubmissionState.Initializing
+        ) {
+          console.error('Attempted to confirm an uninitialized limit order')
+        } else {
+          console.error('Attempted to confirm an in-progress limit order')
+        }
+        return
+      }
+
+      const allowanceResetRequired =
+        state.orderSubmission[limitOrderQuoteId].allowanceReset.isRequired
+      const approvalRequired =
+        state.orderSubmission[limitOrderQuoteId].allowanceApproval.isInitiallyRequired
+
+      switch (true) {
+        case allowanceResetRequired:
+          state.orderSubmission[limitOrderQuoteId].state =
+            LimitOrderSubmissionState.AwaitingAllowanceReset
+          break
+        case approvalRequired:
+          state.orderSubmission[limitOrderQuoteId].state =
+            LimitOrderSubmissionState.AwaitingAllowanceApproval
+          break
+        default:
+          state.orderSubmission[limitOrderQuoteId].state =
+            LimitOrderSubmissionState.AwaitingLimitOrderSubmission
+          break
+      }
+    },
+    setAllowanceResetTxPending: (state, action: PayloadAction<QuoteId>) => {
+      const id = action.payload
+      state.orderSubmission[id].allowanceReset.state = TransactionExecutionState.Pending
+    },
+    setAllowanceApprovalTxPending: (state, action: PayloadAction<QuoteId>) => {
+      const id = action.payload
+      state.orderSubmission[id].allowanceApproval.state = TransactionExecutionState.Pending
+    },
+    setAllowanceResetTxFailed: (state, action: PayloadAction<QuoteId>) => {
+      const id = action.payload
+      state.orderSubmission[id].allowanceApproval.state = TransactionExecutionState.Failed
+    },
+    setAllowanceApprovalTxFailed: (state, action: PayloadAction<QuoteId>) => {
+      const id = action.payload
+      state.orderSubmission[id].allowanceApproval.state = TransactionExecutionState.Failed
+    },
+    setAllowanceResetTxComplete: (state, action: PayloadAction<QuoteId>) => {
+      const id = action.payload
+      state.orderSubmission[id].allowanceReset.state = TransactionExecutionState.Complete
+    },
```

Comment:
Yes its actually insane. The idea here is to sacrifice verbosity here for our own sanity in the UI where tracking the state transitions will get gnarly. This is a near-clone of the model we're using for spot trades, though I can see it's probably just as clear to have `setAllowanceResetTx(TransactionExecutionState.Complete)` instead of `setAllowanceResetTxComplete()`.

If you feel strongly we can change it, but keep in mind it'd change this to being nearly identical to spot trades, to being a tiny bit different.

Reply:
I don't mind keeping them for the sake of PR diff and code consistency